.github/workflows/frontend-tests.yml

```diff
@@ -0,0 +1,71 @@
+name: Frontend Tests
+
+on:
+  pull_request:
+    paths:
+      - '**/frontend/**'
+      - '!frontend/test/e2e/**'
+      - '!**/*.cy.*'
+      - '!**/*cypress*'
+      - '!**/*.md'
+      - 'shared/**'
+      - '**/package.json'
+      - '**/yarn.lock'
+      - '.github/workflows/**'
+  push:
+    branches:
+      - master
+      - 'release**'
+      - 'feature**'
+      - 'fix**'
+      - 'ci**'
+    tags:
+      - '**'
+    paths:
+      - '**/frontend/**'
+      - '!frontend/test/e2e/**'
+      - '!**/*.cy.*'
+      - '!**/*cypress*'
+      - '!**/*.md'
+      - 'shared/**'
+      - '**/package.json'
+      - '**/yarn.lock'
+      - '.github/workflows/**'
```

Comment:
_In theory_, we possibly don't even need to include these conditional paths for `push` events. That is - we wouldn't need them had we restricted branches to `master` and `release**` only.
In that scenario `push` event couldn't possibly be triggered before the successful PR merge.

It is quite possible (and not that uncommon) that people name their branches `fix-something`, or to match `feature**` / `ci**` pattern. In such cases, it is still better to have the conditional path checks. It might seem a bit redundant but it definitely covers some edge cases.